.github/workflows/caching.yml

```diff
@@ -55,42 +67,45 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-        with:
-          submodules: true
+
       - uses: haskell/actions/setup@v1
         with:
-          ghc-version  : ${{ matrix.ghc   }}
+          ghc-version: ${{ matrix.ghc }}
           cabal-version: ${{ matrix.cabal }}
 
-      - if: matrix.os == 'windows-latest'
-        name: Set some window specific things
+      - if: runner.os == 'Windows'
+        name: (Windows) Platform config
         run: |
           echo "CABAL_STORE_DIR=$SYSTEMDRIVE\\SR" >> $GITHUB_ENV
           echo "CABAL_PKGS_DIR=~\\AppData\\cabal\\packages" >> $GITHUB_ENV
-
-      - if: matrix.os != 'windows-latest'
-        name: Set some linux/macOS specific things
+      - if: ( runner.os == 'Linux' ) || ( runner.os == 'macOS' )
+        name: (Linux,macOS) Platform config
         run: |
           echo "CABAL_STORE_DIR=~/.cabal/store" >> $GITHUB_ENV
           echo "CABAL_PKGS_DIR=~/.cabal/packages" >> $GITHUB_ENV
 
-      - if: matrix.os == 'macOS-latest' && matrix.ghc == '8.10.5'
-        name: Workaround for GHC 8.10.5 on macOS
-        run: |
-          echo "# uninstalling CommandLineTools (see https://github.com/haskell/haskell-language-server/issues/1913#issuecomment-861667786)"
-          sudo rm -rf /Library/Developer/CommandLineTools
-
       # Needs to be before Cache Cabal so the cache can detect changes to the modified cabal.project file
       - if: matrix.ghc == '9.0.1'
-        name: Use modified cabal.project for ghc9
-        run: cp cabal-ghc901.project cabal.project
-
-      - if: matrix.ghc == '8.8.4' && matrix.os == 'windows-latest'
-        name: Modify cabal.project to workaround segfaults for ghc-8.8.4 and windows
+        name: (GHC 9.0.1) Use modified `cabal.project`
+        run: |
+          cp cabal-ghc901.project cabal.project
+      - if: runner.os == 'Windows' && matrix.ghc == '8.8.4'
+        name: (Windows,GHC 8.8) Modify `cabal.project` to workaround segfaults
         run: |
           echo "package floskell" >> cabal.project
           echo "  ghc-options: -O0" >> cabal.project
 
+      # Shorten binary names as a workaround for filepath length limits in Windows,
```

Comment:
Too long to explain.

It is connected with that builds & runs that use the cache & the cache building - should share the same project setup. And with that that further down the line the full build would be run in caching, because Cabal simply does not allow otherwise (so far does not support that command, to build dependencies of all targets).

Reply:
& since Cabal provides an interface to get configuration information - I am not sure that executable names are not involved in use somewhere in the code, so I just preserved the same bootstrapping.

Reply:
Removed it.

Reply:
In fact, this code should been there all along.

Because the caching key defined through: `-${{ hashFiles('cabal.project') }}`
& that live patching actively:
```yaml
          sed -i.bak -e 's/haskell-language-server/hls/g' \
                     -e 's/haskell_language_server/hls/g' \
                     haskell-language-server.cabal cabal.project
          sed -i.bak -e 's/Paths_haskell_language_server/Paths_hls/g' \
                     src/**/*.hs exe/*.hs
```

Changes the `cabal.project`, which means that cache does not get hit directly & so workflows that have different `cabal.project` store the same hash as many times.